cmake/kokkos_tribits.cmake

```diff
@@ -238,12 +238,25 @@ FUNCTION(KOKKOS_SET_LIBRARY_PROPERTIES LIBRARY_NAME)
     ""
     ${ARGN})
 
-  IF(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.13")
-    #great, this works the "right" way
+  IF(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.18")
+    #I can use link options
+    #check for CXX linkage using the simple 3.18 way
     TARGET_LINK_OPTIONS(
-      ${LIBRARY_NAME} PUBLIC ${KOKKOS_LINK_OPTIONS}
+      ${LIBRARY_NAME} PUBLIC
+      $<$<LINK_LANGUAGE:CXX>:${KOKKOS_LINK_OPTIONS}>
+    )
+  ELSEIF(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.13")
+    #I can use link options
+    #check for CXX linkage in a super complicated way
+    SET(IS_STATIC $<STREQUAL:$<TARGET_PROPERTY:TYPE>,STATIC_LIBRARY>)
+    #I can't query linker_language on static libraries (and it doesn't matter anyway)
+    SET(LINK_TYPE $<${IS_STATIC}:CXX>$<$<NOT:${IS_STATIC}>:$<TARGET_PROPERTY:LINKER_LANGUAGE>>)
+    TARGET_LINK_OPTIONS(
+      ${LIBRARY_NAME} PUBLIC
+      $<$<STREQUAL:${LINK_TYPE},CXX>:${KOKKOS_LINK_OPTIONS}>
```

Comment:
An alternative would be we only support this with CMake 3.18

Reply:
We will discuss this today, but I am also in favor of dropping this elseif branch and converting it back to what it was before. I.e. only support the CXX only linkage with cmake 3.18.